lib/rabbitmq/cli/core/validators.ex

```diff
@@ -17,14 +17,16 @@
 # Small helper functions, mostly related to connecting to RabbitMQ and
 # handling memory units.
 
-defmodule RabbitMQ.CLI.Ctl.Validators do
+defmodule RabbitMQ.CLI.Core.Validators do
   alias RabbitMQ.CLI.Core.Helpers, as: Helpers
 
-
   def chain([validator | rest], args) do
     case apply(validator, args) do
-      :ok -> chain(rest, args);
-      err -> err
+      :ok                        -> chain(rest, args)
+      {:ok, _}                   -> chain(rest, args)
+      {:validation_failure, err} -> {:validation_failure, err}
+      {:error, err}              -> {:validation_failure, err}
+      other                      -> other
```

Comment:
It could be just me, but I would prefer validation functions to be strictly typed. Otherwise how do we know that `handle_command_output` can handle the return value?